internal/checks/checks_test.go

```diff
@@ -345,17 +415,19 @@ func TestCheckHandlerProbeValidation(t *testing.T) {
 		},
 	}
 
-	for _, tc := range testcases {
-		u, err := NewUpdater(tc.opts)
-		require.NoError(t, err)
+	for testName, tc := range testcases {
+		t.Run(testName, func(t *testing.T) {
```

Comment:
If we're feeling fancy we might as well add `t.Parallel()` here (with the usual copy guard)

```suggestion
	for testName, tc := range testcases {
		tc := tc
			
		t.Run(testName, func(t *testing.T) {
		t.Parallel()
		
```

Reply:
Absolutely, see [bcacc34](https://github.com/grafana/synthetic-monitoring-agent/pull/809/commits/bcacc344a8ab931546875192b7e7bcc8465652f8).
I think we should iteratively modify every test and subest to run in parallel, or at some point dedicate some time for a big PR.

Reply:
@ka3de Do mind the `tc := tc` line, otherwise the test will break badly and cryptically as goroutines start to share a variable that changes across iterations.

Reply:
Yeah, I usually forget that they did the trick for us and write the `tc := tc` anyway, but I think it's time to not do it :smile: